File: 2021/fsharp/Aoc2021.Tests/Day06.Tests.fs
module Day06.Tests

open Xunit

let input = [ 3; 4; 3; 1; 2 ]

[<Fact>]
let ``Day06 simulate works`` () =
    Assert.Equal(26L, input |> simulateFishPopulation |> Seq.item 18)
    Assert.Equal(5934L, input |> simulateFishPopulation |> Seq.item 80)
    Assert.Equal(26984457539L, input |> simulateFishPopulation |> Seq.item 256)

[<Fact>]
let ``Day06 Stars`` () =
    Assert.Equal("350605", job1 ())
    Assert.Equal("1592778185024", job2 ())

File: 2021/csharp/Aoc2021.Tests/Day06Tests.cs
namespace Aoc2021.Tests;

public class Day06Tests
{
    private static readonly List<long> input = "3,4,3,1,2"
        .Split(',').Select(Utils.ParseLong).ToList();

    [Fact]
    public void SimulateWorks()
    {
        Assert.Equal(26, Day06.SimulateFishPopulation(input).ElementAt(18));
        Assert.Equal(5934, Day06.SimulateFishPopulation(input).ElementAt(80));
        Assert.Equal(26984457539, Day06.SimulateFishPopulation(input).ElementAt(256));
    }

    [Fact]
    public void Stars()
    {
        Day06 run = new();
        Assert.Equal("350605", run.Result1());
        Assert.Equal("1592778185024", run.Result2());
    }
}
